cluster-autoscaler/core/scale_down.go

```diff
@@ -146,6 +177,58 @@ func calculateCoresAndMemoryTotal(nodes []*apiv1.Node, timestamp time.Time) (int
 	return coresTotal, memoryTotal
 }
 
+func calculateGpusTotal(nodes []*apiv1.Node, cp cloudprovider.CloudProvider, timestamp time.Time) (map[string]int64, error) {
+	type gpuInfo struct {
+		name  string
+		count int64
+	}
+
+	result := make(map[string]int64)
+	ngCache := make(map[string]gpuInfo)
+	for _, node := range nodes {
+		if isNodeBeingDeleted(node, timestamp) {
+			// Nodes being deleted do not count towards total cluster resources
+			continue
+		}
+		nodeGroup, err := cp.NodeGroupForNode(node)
+		if err != nil {
+			return nil, errors.ToAutoscalerError(errors.CloudProviderError, err).AddPrefix("can not get node group for node %v when calculating cluster gpu usage", node.Name)
+		}
+		if reflect.ValueOf(nodeGroup).IsNil() {
+			// TODO[lukaszos] why do we need this? do we?
+			nodeGroup = nil
+		}
+
+		var gpuType string
+		var gpuCount int64
+
+		cacheHit := false
+		if nodeGroup != nil {
+			cached, found := ngCache[nodeGroup.Id()]
```

Comment:
Either cacheHit or found is redunant, e.g.:
```cached, cacheHit := ngCache[nodeGroup.Id()]```

Reply:
Nevermind, the scope doesn't work out.

Reply:
I usually deal with that by declaring both variables and avoiding the pure, vile evil that is := with multiple assignement.
```
var cached, found bool
if stuff() {
  cached, found = whatever()
}
```
I'm fine with @losipiuk's solution too. Also - operator :=, scopes and multiple assignement is the single largest design flow of golang :(

Reply:
Changed